cpp/util/algorithm.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2015-2015 the openage authors. See copying.md for legal info.
+#ifndef OPENAGE_UTIL_ALGORITHM_H_
+#define OPENAGE_UTIL_ALGORITHM_H_
+
+#include <algorithm>
+
+// TODO: make general transformation from iterator algorithm to container
+// algorithm. so much metaprogramming...
+
+namespace openage {
+namespace util {
+
+/**
+ * std::for_each except just on containers.
+ */
+template<class Container, class Function>
+inline Function for_each(Container &&container, Function &&func) {
+	// why cpp why...
+	return std::for_each(std::begin(std::forward<Container>(container)),
+	                     std::end(std::forward<Container>(container)),
+	                     std::forward<Function>(func));
+}
+
+/**
+ * Filters items from a container which satisfy a certain predicate.
+ */
+template<class Container, class Function>
+inline void remove_from(Container &container, Function &&func) {
+	container.erase(std::remove_if(std::begin(container),
+	                               std::end(container),
+	                               std::forward<Function>(func)));
```

Comment:
Be VERY careful here. I don't think this code is doing what you think it does! This will only erase ONE element from the container if the predicate is satisfied at least once. If more than one element satisfies the predicate the remaining elements would be moved at the end of the container and will remain there. If no element satisfies the predicate this code would try to erase std::end(container) which is an undefined behavior!

The correct code would be : 
container.erase(std::remove_if(std::begin(container),
	                               std::end(container),
	                               std::forward<Function>(func))
                          std::end(container)); // Note the additional parameter here!
This would call the correct erase overload for an std::vector but unfortunately this would not work for std::list, std::map and other containers and there is no uniform way (yet) to do this for all containers.
For more details : 
- http://stackoverflow.com/questions/16013545/how-do-i-erase-elements-from-stl-containers
- http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4161.htm